### DISCUSSION TITLE
Warn if Pydantic is changing types

### DISCUSSION THREAD

**Discussion by AmirHmZz** [pydantic/pydantic#4276] [Question]
Consider I have a model like this :
```python
class model(BaseModel):
    a: int
```

I want to Pydantic warn me if it is helping me in changing types. I mean if I did something like this : 
```python
model(**{"a":"3"})
```
I except Pydantic creates the model for me but also warns me that type of `"a"` is not `integer`. How can I achieve something like this ?

**Comment by samuelcolvin** [comment]
Short answer, no. I'm afraid not really, you could add a pre-validator and try to detect the wrong type and print/store the warning somewhere global.

In V2 you'll be able to do it, albeit with a dedicated validator function for each field.

**Reply by AmirHmZz** [reply]
Thank you for the answer and thanks a lot for developing Pydantic :)
